RadioIDs.py

```diff
@@ -0,0 +1,48 @@
+import logging
+from typing import Dict, List
+
+log = logging.getLogger(__name__)
+namecache: Dict[str, str] = {}
+
+
+def _getSet(srcList: List[dict]) -> List[str]:
+    """Generates a list of unique sources.
+
+    Args:
+        srcList (List[dict]): The srcList to process.
+
+    Returns:
+        List[str]: Unique source IDs.
+    """
+    result = set()
+    for src in srcList:
+        result.add(src["src"])
+    return list(result)
+
+
+def _scrape(sources: List[str]) -> List[str]:
+    """[summary]
+
+    Args:
+        sources (List[str]): [description]
+
+    Returns:
+        List[str]: [description]
+    """
+    names: List[str] = []
+    for source in sources:
+        if source in namecache.keys:
```

Comment:
I think this might need to be either `if source in namecache` or `if source in namecache.keys()` (both are the same IIRC, but `keys` is a list so if it's reference it has to be called)